examples/rag/rag_self_correction.py

```diff
@@ -98,73 +75,78 @@ def run(
         return {"documents": next_page}
 
 
-# Create the RAG pipeline
-rag_pipeline = Pipeline(max_loops_allowed=10)
-rag_pipeline.add_component(instance=Switch(type_=str), name="query")
-rag_pipeline.add_component(
-    instance=PaginatedRetriever(InMemoryBM25Retriever(document_store=InMemoryDocumentStore())), name="retriever"
-)
-rag_pipeline.add_component(
-    instance=PromptBuilder(
-        template="""
-Given these documents, answer the question.
-If the documents don't provide enough information to answer the question, answer with the string "UNKNOWN".
-
-Documents:
-{% for doc in documents %}
-    {{ doc.content }}
-{% endfor %}
-
-Question: {{question}}
-Answer:
-"""
-    ),
-    name="prompt_builder",
-)
-rag_pipeline.add_component(instance=GPTGenerator(), name="llm")
-rag_pipeline.add_component(
-    instance=ConditionalRouter(
-        routes=[
-            {
-                "condition": "{{ 'UNKNOWN' in replies|join(' ') }}",
-                "output": "{{ query }}",
-                "output_name": "unanswered_query",
-                "output_type": str,
-            },
-            {
-                "condition": "{{ 'UNKNOWN' not in replies|join(' ') }}",
-                "output": "{{ replies }}",
-                "output_name": "replies",
-                "output_type": List[str],
-            },
-        ]
-    ),
-    name="answer_checker",
-)
-
-rag_pipeline.connect("query", "retriever")
-rag_pipeline.connect("query", "prompt_builder.question")
-rag_pipeline.connect("query", "answer_checker.query")
-rag_pipeline.connect("retriever", "prompt_builder.documents")
-rag_pipeline.connect("prompt_builder", "llm")
-rag_pipeline.connect("llm.replies", "answer_checker.replies")
-rag_pipeline.connect("answer_checker.unanswered_query", "query")
-
-# Draw the pipeline
-rag_pipeline.draw("test_bm25_rag_pipeline.png")
-
-# Populate the document store
-documents = [
-    Document(content="My name is Jean and I live in Paris."),
-    Document(content="My name is Mark and I live in Berlin."),
-    Document(content="My name is Giorgio and I live in Rome."),
-    Document(content="My name is Juan and I live in Madrid."),
-]
-rag_pipeline.get_component("retriever").retriever.document_store.write_documents(documents)
-
-# Query and assert
-question = "Who lives in Germany?"
-
-result = rag_pipeline.run({"query": {"value": question}})
-
-pprint(result)
+def self_correcting_pipeline():
+    # Create the RAG pipeline
+    rag_pipeline = Pipeline(max_loops_allowed=10)
+    rag_pipeline.add_component(instance=Multiplexer(str), name="query")
```

Comment:
```suggestion
    rag_pipeline.add_component(instance=Multiplexer(str), name="query_multiplexer")
```

What about a more explicit name?